fw/cache.c

```diff
@@ -2290,9 +2298,12 @@ tfw_cache_do_action(TfwHttpMsg *msg, tfw_http_cache_cb_t action)
 	}
 
 	req = (TfwHttpReq *)msg;
-	if (unlikely(req->method == TFW_HTTP_METH_PURGE))
-		tfw_cache_purge_method(req);
-	else
+	if (unlikely(req->method == TFW_HTTP_METH_PURGE && !TFW_MSG_H2(req))) {
```

Comment:
I don't think we need an explicit check here - what's wrong if we receive PURGE on h2? Probably it's better just to remove the method processing from h2 parsing - the parser will be faster and we'll need to support less code.